### ISSUE TITLE
USB Redirection of Android device not working

### ISSUE THREAD

**Issue by meanbunny** [FreeRDP/FreeRDP#2742]
Hey all,

I am trying to redirect my Android device (Galaxy Nexus 6) to my Windows 8.1 VM. I am currently running Ubuntu 15.04 and have pulled the latest source code and built. I have also installed all of the needed libraries in order to get the URBDRC to at least be built. Below are the errors that I am receiving.

```
[16:06:10:759] [3941:3942] [INFO][com.freerdp.client.common.cmdline] - loading channel rdpdr
[16:06:10:759] [3941:3942] [INFO][com.freerdp.client.common.cmdline] - loading channel rdpsnd
[16:06:10:759] [3941:3942] [INFO][com.freerdp.client.common.cmdline] - loading channel cliprdr
[16:06:10:759] [3941:3942] [INFO][com.freerdp.client.common.cmdline] - loading channel drdynvc
[16:06:10:621] [3941:3942] [INFO][com.freerdp.channels.drdynvc.client] - Loading Dynamic Virtual Channel urbdrc
[16:06:10:626] [3941:3942] [WARN][com.freerdp.channels.urbdrc.client] - bus:0 dev:0 not exist in udevman
[16:06:10:626] [3941:3942] [INFO][com.freerdp.channels.urbdrc.client] - VID: 0x18D1, PID: 0x4EE2
[16:06:10:630] [3941:3942] [ERROR][com.freerdp.channels.urbdrc.client] - USB init: Error to get HUB handle!!
[16:06:10:630] [3941:3942] [WARN][com.freerdp.channels.urbdrc.client] - bus:1 dev:8 not exist in udevman
[16:06:10:630] [3941:3942] [INFO][com.freerdp.channels.drdynvc.client] - Loading Dynamic Virtual Channel audin
```

My command looks like this. 

```
xfreerdp /title:mm-local-desk2 +wallpaper +clipboard /usb:id,dev:18d1:4ee2 /sound:sys:pulse /microphone:sys:pulse /h:872 /w:1600 /v:192.168.122.110
```

Can anyone point me in the right direction in code where I could start working on this?

**Comment by akallabeth** [comment]
@meanbunny Please check the access rights for your USB devices. Looks like you are missing some privileges and libusb can not access the devices.

**Comment by meanbunny** [comment]
I just ran it under `sudo su` is that enough? The error remained. The usb device that I am trying to attach to shows up with `lsusb` command

Bus 004 Device 002: ID 8087:0024 Intel Corp. Integrated Rate Matching Hub
Bus 004 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 003 Device 004: ID 1bcf:2885 Sunplus Innovation Technology Inc. ASUS Webcam
Bus 003 Device 005: ID 13d3:3362 IMC Networks 
Bus 003 Device 002: ID 8087:0024 Intel Corp. Integrated Rate Matching Hub
Bus 003 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 006: ID 18d1:4ee2 Google Inc. Nexus 4 (debug)
Bus 001 Device 005: ID 045e:07a5 Microsoft Corp. 
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub

**Comment by HenryJacques** [comment]
@meanbunny : did you enable RemoteFX on the Windows 8.1 VM (in local computer group policies)?
It is mandatory to redirect USB devices such as your phone.
(see https://technet.microsoft.com/fr-fr/library/Ff817578%28v=WS.10%29.aspx)

**Comment by HenryJacques** [comment]
see also #2832

**Comment by akallabeth** [comment]
Fixed with #5745
